metrics is not cleaned up when the owner suicide

### What did you do?

nothing, but find some logs
```
[2022/10/08 16:26:59.400 +08:00] [INFO] [client.go:235] ["WatchWithChan exited"] [role=owner]
[2022/10/08 16:26:59.400 +08:00] [INFO] [client.go:235] ["WatchWithChan exited"] [role=processor]
[2022/10/08 16:26:59.400 +08:00] [WARN] [capture.go:500] ["session is disconnected"] [error="[CDC:ErrEtcdSessionDone]the etcd session is done"] [errorVerbose="[CDC:ErrEtcdSessionDone]the etcd session is done\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run\n\tgithub.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:186\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:491\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).campaignOwner\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:443\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:313\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
[2022/10/08 16:26:59.400 +08:00] [WARN] [capture.go:500] ["session is disconnected"] [error="[CDC:ErrEtcdSessionDone]the etcd session is done"] [errorVerbose="[CDC:ErrEtcdSessionDone]the etcd session is done\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/orchestrator.(*EtcdWorker).Run\n\tgithub.com/pingcap/tiflow/pkg/orchestrator/etcd_worker.go:186\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:491\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func4\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:334\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
[2022/10/08 16:26:59.401 +08:00] [INFO] [capture.go:335] ["processor routine exited"] [captureID=b1b70b7e-4d86-4507-9074-299866278b2e] [error="[CDC:ErrCaptureSuicide]capture suicide"] [errorVerbose="[CDC:ErrCaptureSuicide]capture suicide\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:501\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func4\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:334\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
[2022/10/08 16:26:59.407 +08:00] [INFO] [worker.go:81] ["FlushWorker exited"] [error="context canceled"] [errorVerbose="context canceled\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15\ngithub.com/pingcap/tiflow/cdc/sinkv2/eventsink/mq.(*worker).run\n\tgithub.com/pingcap/tiflow/cdc/sinkv2/eventsink/mq/worker.go:92\ngithub.com/pingcap/tiflow/cdc/sinkv2/eventsink/mq.newSink.func1\n\tgithub.com/pingcap/tiflow/cdc/sinkv2/eventsink/mq/mq_dml_sink.go:89\nruntime.goexit\n\truntime/asm_amd64.s:1571"] [namespace=default] [changefeed=canal-json-kafka-test]
[2022/10/08 16:26:59.492 +08:00] [INFO] [client.go:297] ["peer-to-peer client retrying"] [topic=changefeed/default/canal-json-kafka-test/scheduler] [fromSeq=1040]
[2022/10/08 16:27:00.486 +08:00] [INFO] [client.go:297] ["peer-to-peer client retrying"] [topic=changefeed/default/canal-json-kafka-test/scheduler] [fromSeq=1040]
[2022/10/08 16:27:00.486 +08:00] [WARN] [server.go:646] ["error receiving from peer"] [error="context canceled"]
[2022/10/08 16:27:01.485 +08:00] [INFO] [client.go:297] ["peer-to-peer client retrying"] [topic=changefeed/default/canal-json-kafka-test/scheduler] [fromSeq=1040]
[2022/10/08 16:27:01.486 +08:00] [WARN] [server.go:646] ["error receiving from peer"] [error="context canceled"]
[2022/10/08 16:27:02.486 +08:00] [INFO] [client.go:297] ["peer-to-peer client retrying"] [topic=changefeed/default/canal-json-kafka-test/scheduler] [fromSeq=1040]
[2022/10/08 16:27:02.792 +08:00] [ERROR] [client.go:910] ["[pd] update connection contexts failed"] [dc=global] [error="rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 10.244.4.132:2379: connect: connection refused\""]
[2022/10/08 16:27:02.827 +08:00] [ERROR] [client.go:910] ["[pd] update connection contexts failed"] [dc=global] [error="rpc error: code = Unavailable desc = connection error: desc = \"transport: Error while dialing dial tcp 10.244.4.132:2379: connect: connection refused\""]
[2022/10/08 16:27:03.485 +08:00] [INFO] [client.go:297] ["peer-to-peer client retrying"] [topic=changefeed/default/canal-json-kafka-test/scheduler] [fromSeq=1040]
[2022/10/08 16:27:04.401 +08:00] [WARN] [capture.go:459] ["owner resign timeout"] [captureID=b1b70b7e-4d86-4507-9074-299866278b2e] [error="[CDC:ErrCaptureResignOwner]resign owner failed: context deadline exceeded"] [errorVerbose="[CDC:ErrCaptureResignOwner]resign owner failed: context deadline exceeded\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/errors.WrapError\n\tgithub.com/pingcap/tiflow/pkg/errors/helper.go:32\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).resign\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:545\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).campaignOwner\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:451\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:313\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1571"] [ownerRev=6617129]
[2022/10/08 16:27:04.402 +08:00] [INFO] [capture.go:464] ["owner resigned successfully"] [captureID=b1b70b7e-4d86-4507-9074-299866278b2e] [ownerRev=6617129]
[2022/10/08 16:27:04.402 +08:00] [WARN] [capture.go:467] ["run owner exited with error"] [captureID=b1b70b7e-4d86-4507-9074-299866278b2e] [ownerRev=6617129] [error="[CDC:ErrCaptureSuicide]capture suicide"] [errorVerbose="[CDC:ErrCaptureSuicide]capture suicide\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:501\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).campaignOwner\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:443\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:313\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
[2022/10/08 16:27:04.402 +08:00] [INFO] [capture.go:314] ["owner routine exited"] [captureID=b1b70b7e-4d86-4507-9074-299866278b2e] [error="[CDC:ErrCaptureSuicide]capture suicide"] [errorVerbose="[CDC:ErrCaptureSuicide]capture suicide\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/normalize.go:164\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).runEtcdWorker\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:501\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).campaignOwner\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:443\ngithub.com/pingcap/tiflow/cdc/capture.(*captureImpl).run.func3\n\tgithub.com/pingcap/tiflow/cdc/capture/capture.go:313\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
[2022/10/08 16:27:04.402 +08:00] [INFO] [capture.go:574] ["processor manager closed"] [captureID=b1b70b7e-4d86-4507-9074-299866278b2e]
[2022/10/08 16:27:04.402 +08:00] [INFO] [client.go:117] ["peer message client exited"] [addr=lingjin-4.jinling.tispace:8300] [captureID=b1b70b7e-4d86-4507-9074-299866278b2e] [error="context canceled"] [errorVerbose="context canceled\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageClient).Run\n\tgithub.com/pingcap/tiflow/pkg/p2p/client.go:139\ngithub.com/pingcap/tiflow/pkg/p2p.(*messageRouterImpl).GetClient.func1\n\tgithub.com/pingcap/tiflow/pkg/p2p/message_router.go:144\nruntime.goexit\n\truntime/asm_amd64.s:1571"]

```

### What did you expect to see?

metrics looks normal

### What did you see instead?

<img width="2931" alt="image" src="https://user-images.githubusercontent.com/7138436/194701841-814dda68-7f7d-4e42-ac0c-6897e13494bf.png">

metrics was not cleaned up 

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
master
```

--- a/cdc/capture/capture.go
+++ b/cdc/capture/capture.go
@@ -445,6 +445,7 @@ func (c *captureImpl) campaignOwner(ctx cdcContext.Context) error {
 		err = c.runEtcdWorker(ownerCtx, owner,
 			orchestrator.NewGlobalState(c.EtcdClient.GetClusterID()),
 			ownerFlushInterval, util.RoleOwner.String())
+		c.owner.AsyncStop()
 		c.setOwner(nil)
 
 		// if owner exits, resign the owner key,

